pkg/chartverifier/certifier.go

```diff
@@ -82,13 +82,17 @@ func (c *certifier) Certify(uri string) (*Certificate, error) {
 	oc := tool.NewOc(procExec)
 
 	osVersion, err := oc.GetVersion()
-	if err != nil && c.openshiftVersion == "" {
-		return nil, OpenShiftVersionErr(err.Error())
+	if err != nil {
+		if c.openshiftVersion == "" {
+			return nil, OpenShiftVersionErr(err.Error())
+		} else {
+			osVersion = c.openshiftVersion
+		}
 	}
-	if _, err := semver.NewVersion(c.openshiftVersion); err != nil {
+
+	if _, err := semver.NewVersion(osVersion); err != nil {
```

Comment:
This change is good, although I believe the assignment to `osVersion` below should be performed before this branch; it also makes the change regarding the err and openshiftVersion predicate not necessary.

Reply:
@isutton Since the logic is a bit difficult to explain. I have extracted a function and written few unit tests. Please take a look.